Query profiles

dffischer opened the pull request:
While using the changes from pull request #2 in practice, I noticed that authors were not always assigned correctly. Because Vine supports reposting, videos of other authors may appear in the examined timeline. So guessing the author of the whole feed from any entry is always prone to fail.

This pull request improves on that by first looking at the profile of the requested ID to collect information concerning the whole feed, Only afterwards it processes the timeline, assigning individual authorship to each posting.

For the cost of the additional request to the profile, the feed can then carry the original description now readily available there.

dffischer committed:
Query feed information from user profile

Querying the user from the first video is not always accurate: It may be
a repost and thus designated to the original author. The profile
information yields definitive information about the user by id.

dffischer committed:
Include profile description

Utilizing more information available in the profile, a sensible
description is readily available.

According to the API description [1] an "email" field should also be
available. However, I could not find this field filled in a sample of
randomly selected Vine users, so I assume it is only visible when logged
in or when examining the own account. It thus an be regarded unfilled
for the purposes of vinefeed.

[1]: https://github.com/starlock/vino/wiki/API-Reference#get-user-data

dffischer committed:
Externalize error handling to function

This reduces code duplication

tremby commented:
I don't have time to look properly at this right now. Just messaging here to acknowledge. I'll get to it when I can!

One quick question, just from my glance over the description: with a large feed with a lot of different reposted authors, would this mean a whole lot of extra requests to the server? Is it much slower? Do we need to worry about any API limits?

dffischer committed:
Parallelize requests

Both request compose different information for the feed, not building up
on each other, so they can run in parallel, sending out the result when
both have finished.

dffischer commented:
Take your time.

As the email fields are always empty, querying each original author of every repost would yield no additional relevant information, so the only profile queried is the one of the feed owner itself. One request (to the timeline) became two requests (to profile and timeline of the same user) - This should not strain the API much, but obviously double request time as long as these happen sequentially. I just added another commit to parallelize them. Now, the process should not even take significantly longer.

tremby commented:
Sounds good. I'll look in more detail when I can.
